### ISSUE TITLE
Fcitx5 not works in some app

### ISSUE THREAD

**Issue by ZreXoc** [hyprwm/Hyprland#6179]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit ed58cc4c31c21e09ee780d0df818afe935181cd9  (sessionLock: remove m_pLastFocus on destroy of surface).
Date: Wed May 1 09:33:36 2024
Tag: v0.39.1-117-ged58cc4c, commits: 4577

flags: (if any)


System Information:
System name: Linux
Node name: xc-arch
Release: 6.8.9-arch1-2
Version: #1 SMP PREEMPT_DYNAMIC Tue, 07 May 2024 21:35:54 +0000


GPU information: 
00:02.0 VGA compatible controller [0300]: Intel Corporation Raptor Lake-S UHD Graphics [8086:a788] (rev 04) (prog-if 00 [VGA controller])
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation AD107M [GeForce RTX 4060 Max-Q / Mobile] [10de:28e0] (rev a1) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
```

</details>

### Bug or Regression?

Bug

### Description

Fcitx5 not works in following apps:
1. Mathematica
2. Matlab
3. all steam games (full screen or not)
I don't know what the common these programs have :(

When focusing it, I can't switch to my input method with hotkey. I can switch it manually but can't type anything

### How to reproduce

Install one of the programs and try typing with fcitx.

### Crash reports, logs, images, videos

fcitx5 version : 5.1.9

Output of `hyprctl clients `:
Mathematica:
```
Window 5cbf4409d3c0 -> Untitled-1 * - Wolfram Mathematica:
	mapped: 1
	hidden: 0
	at: 1274,864
	size: 1276,532
	workspace: 1 (1)
	floating: 0
	monitor: 1
	class: com.wolfram.Mathematica.14.0
	title: Untitled-1 * - Wolfram Mathematica
	initialClass: com.wolfram.Mathematica.14.0
	initialTitle: Untitled-1 - Wolfram Mathematica
	pid: 262689
	xwayland: 0
	pinned: 0
	fullscreen: 0
	fullscreenmode: 0
	fakefullscreen: 0
	grouped: 0
	swallowing: 0
	focusHistoryID: 2
```
matlab:
```
Window 5cbf431073f0 -> matlab:
	mapped: 1
	hidden: 1
	at: 10,10
	size: 861,1386
	workspace: 1 (1)
	floating: 0
	monitor: 1
	class: kitty
	title: matlab
	initialClass: kitty
	initialTitle: zsh
	pid: 291471
	xwayland: 0
	pinned: 0
	fullscreen: 0
	fullscreenmode: 0
	fakefullscreen: 0
	grouped: 0
	swallowing: 0
	focusHistoryID: 3

Window 5cbf43f43600 -> MATLAB R2023b:
	mapped: 1
	hidden: 0
	at: -615,223
	size: 1271,1019
	workspace: 1 (1)
	floating: 1
	monitor: 1
	class: MATLAB R2023b
	title: MATLAB R2023b
	initialClass: MATLAB R2023b
	initialTitle: MATLAB R2023b
	pid: 291652
	xwayland: 1
	pinned: 0
	fullscreen: 0
	fullscreenmode: 0
	fakefullscreen: 0
	grouped: 0
	swallowing: 5cbf431073f0
	focusHistoryID: 2
```

**Comment by vaxerski** [comment]
does it work on sway?

**Comment by qomosoloto** [comment]
Got same issue suddenlly, don't know why. Maybe caused by some updates today (on archlinux), and `bind = Ctrl+Shift, Home, exec, pkill fcitx5 || fcitx5 # restart fcitx5` will crash hyprland to logout which never happend before!!! And I can not use fcitx5 in obsidian since today, I can switch it manually but can't type anything. But it's okay before today.

**Comment by vaxerski** [comment]
provide a debug stacktrace if crashing

**Comment by ZreXoc** [comment]
In my condition fcitx didn't crash, just can't input in certain programs. Sometimes the candidate box will displays on the top left corner of the screen, but when I select the word, nothing was input.

**Comment by ZreXoc** [comment]
> Got same issue suddenlly, don't know why. Maybe caused by some updates today (on archlinux), and `bind = Ctrl+Shift, Home, exec, pkill fcitx5 || fcitx5 # restart fcitx5` will crash hyprland to logout which never happend before!!! And I can not use fcitx5 in obsidian since today, I can switch it manually but can't type anything. But it's okay before today.
> 
> [hyprlandCrashReport79754.txt](https://github.com/hyprwm/Hyprland/files/15494954/hyprlandCrashReport79754.txt)

your first issue seems to be dupe of https://github.com/hyprwm/Hyprland/issues/2146

**Comment by mjkx5** [comment]
It occured on my google-chrome, opera.

**Comment by qomosoloto** [comment]
> In my condition fcitx didn't crash, just can't input in certain programs. Sometimes the candidate box will displays on the top left corner of the screen, but when I select the word, nothing was input.

Yeah, exactly same. But it worked before, such as Jetbrains IDEs, Obsidian. Saddly now I can't not input chinese when using them...

**Comment by qomosoloto** [comment]
> > Got same issue suddenlly, don't know why. Maybe caused by some updates today (on archlinux), and `bind = Ctrl+Shift, Home, exec, pkill fcitx5 || fcitx5 # restart fcitx5` will crash hyprland to logout which never happend before!!! And I can not use fcitx5 in obsidian since today, I can switch it manually but can't type anything. But it's okay before today.
> > [hyprlandCrashReport79754.txt](https://github.com/hyprwm/Hyprland/files/15494954/hyprlandCrashReport79754.txt)
> 
> your first issue seems to be dupe of #2146

Ant it seem that this problem has not been solved.

**Comment by mjkx5** [comment]
use hyprland-git replace hyprland on archlinux, which reduce probability of the error occur. I donot know why, but It occur in my computer.

**Comment by vaxerski** [comment]
has it ever worked? is this a regression? (aka can you test 0.39.1)

**Comment by ZreXoc** [comment]
> has it ever worked? is this a regression? (aka can you test 0.39.1)

Both not. Actually i got the issue few months ago when i first installed and played a game on hyprland. Except the program i mentioned above, everything have been working fine

**Comment by qomosoloto** [comment]
> use hyprland-git replace hyprland on archlinux, which reduce probability of the error occur. I donot know why, but It occur in my computer.
> 
> It exists, but reduce the frequency of the error.

After switching to hyprland-git, fcitx5 got working !!!!!

**Comment by ZreXoc** [comment]
> > use hyprland-git replace hyprland on archlinux, which reduce probability of the error occur. I donot know why, but It occur in my computer.
> > It exists, but reduce the frequency of the error.
> 
> After switching to hyprland-git, fcitx5 got working !!!!!

version: `hyprland-git 0.40.0.r141.722d537a-1` not work for me. Any config about fcitx5?

**Comment by ZreXoc** [comment]
> > > > use hyprland-git replace hyprland on archlinux, which reduce probability of the error occur. I donot know why, but It occur in my computer.
> > > > It exists, but reduce the frequency of the error.
> > > 
> > > 
> > > After switching to hyprland-git, fcitx5 got working !!!!!
> > 
> > 
> > version: `hyprland-git 0.40.0.r141.722d537a-1` not work for me. Any config about fcitx5?
> 
> Just something in `/etc/environment`, because gnome DE also installed.
> 
> ```
> #
> # This file is parsed by pam_env module
> #
> # Syntax: simple "KEY=VAL" pairs on separate lines
> #
> GTK_IM_MODULE=fcitx
> QT_IM_MODULE=fcitx
> XMODIFIERS=@im=fcitx
> SDL_IM_MODULE=fcitx
> INPUT_METHOD=fcitx
> GLFW_IM_MODULE=ibus
> ```

I didn't set it due to some warnings before. Now I add it back but still not work. : (

**Comment by dingiso** [comment]
I'm having a similar issue. From my experience, all X Window applications (xwayland) don't work with Fcitx5 initially, but the problem can be resolved by restarting the Fcitx5 input method

**Comment by The-Lost-Light** [comment]
Run Appflowy in wayland can't work with Fcitx5.
However after open Appflowy then run `hyprctl reload` then fcitx5 work.
(restarting Fcitx5 changed nothing)

For this I comment all the settings, but still need run the command. So its not relate to my config.
Maybe `hyprctl reload` do something magic!

**Comment by DirkFi** [comment]
Yeah. Same issue here. I can use fcitx5 in most apps but the system google search and cider. The env variables are correct. Still have no clue how to fix it.

**Comment by shootingst4r** [comment]
Installing fcitx5-gtk solved it for me (combined with the env var) (you'll need to restart fcitx5 and the application tho

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
